presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -58,23 +65,37 @@ public WindowOperatorFactory(
                 List<Integer> outputChannels,
                 List<WindowFunctionDefinition> windowFunctionDefinitions,
                 List<Integer> partitionChannels,
+                List<Integer> preGroupedChannels,
                 List<Integer> sortChannels,
                 List<SortOrder> sortOrder,
+                int preSortedChannelPrefix,
                 FrameInfo frameInfo,
                 int expectedPositions)
         {
+            checkNotNull(sourceTypes, "sourceTypes is null");
+            checkNotNull(outputChannels, "outputChannels is null");
+            checkNotNull(windowFunctionDefinitions, windowFunctionDefinitions);
+            checkNotNull(partitionChannels, "partitionChannels is null");
+            checkNotNull(preGroupedChannels, "preGroupedChannels is null");
+            checkArgument(partitionChannels.containsAll(preGroupedChannels), "preGroupedChannels must be a subset of partitionChannels");
+            checkNotNull(sortChannels, "sortChannels is null");
+            checkNotNull(sortOrder, "sortOrder is null");
+            checkArgument(sortChannels.size() == sortOrder.size(), "Must have same number of sort channels as sort orders");
+            checkArgument(preSortedChannelPrefix <= sortChannels.size(), "Cannot have more pre-sorted channels than specified sorted channels");
+            checkArgument(preSortedChannelPrefix == 0 || ImmutableSet.copyOf(preGroupedChannels).equals(ImmutableSet.copyOf(partitionChannels)), "preSortedChannelPrefix can only be greater than zero if all partition channels are pre-grouped");
+            checkNotNull(frameInfo, "frameInfo is null");
```

Comment:
If you are going to rewrite all of the null checks you should switch to `Objects.requireNonNull` at the same time.